src/dmd/lexer.d

```diff
@@ -1635,6 +1637,9 @@ class Lexer
         uint nest = 1;
         const start = loc();
         const pstart = ++p;
+        const oldHandlePoundLine = handlePoundLine;
+        handlePoundLine = false; // #line tokens in token strings have no effect
+        scope(exit) handlePoundLine = oldHandlePoundLine;
```

Comment:
Don't use scope(exit) because it builds an exception handling frame, slowing things down, unless it's all nothrow.

Reply:
Even simpler is just increment inTokenStringConstant on entry and decrement it on exit.

Reply:
The Lexer class as well as all free functions in the module are nothrow.